core/src/main/scala/org/apache/spark/util/collection/ExternalSorter.scala

```diff
@@ -113,11 +114,21 @@ private[spark] class ExternalSorter[K, V, C](
     if (shouldPartition) partitioner.get.getPartition(key) else 0
   }
 
+  private val metaInitialRecords = conf.getInt("spark.shuffle.sort.metaInitialRecords", 256)
+  private val kvChunkSize = conf.getInt("spark.shuffle.sort.kvChunkSize", 8192)
+  private val useSerializedPairBuffer =
+    !ordering.isDefined && conf.getBoolean("spark.shuffle.sort.serializeMapOutputs", true) &&
```

Comment:
I think we'd probably need to ship this as off-by-default for one version and ask people to experiment with it. We've been doing this for most shuffle changes recently.